tools/memap.py

```diff
@@ -98,44 +145,44 @@ def check_new_section_gcc(self, line):
         else:
             return False         # everything else, means no change in section
 
-    
-    def path_object_to_module_name(self, txt):
-        """ Parse a path to object file to extract it's module and object data
+
+    def parse_object_name_gcc(self, line):
+        """ Parse a path to object file
 
         Positional arguments:
         txt - the path to parse the object and module name from
         """
 
-        txt = txt.replace('\\', '/')
-        rex_mbed_os_name = r'^.+mbed-os\/(.+)\/(.+\.o)$'
-        test_rex_mbed_os_name = re.match(rex_mbed_os_name, txt)
+        line = line.replace('\\', '/')
+        rex_mbed_os_name = r'^.+\/(.+\.o)$'
+        test_rex_mbed_os_name = re.match(rex_mbed_os_name, line)
 
         if test_rex_mbed_os_name:
 
-            object_name = test_rex_mbed_os_name.group(2)
-            data = test_rex_mbed_os_name.group(1).split('/')
-            ndata = len(data)
+            object_name = test_rex_mbed_os_name.group(1)
 
-            if ndata == 1:
-                module_name = data[0]
-            else:
-                module_name = data[0] + '/' + data[1]
+            # corner case: certain objects are provided by the GCC toolchain
+            if 'arm-none-eabi' in line:
+                object_name = '[lib]/misc/' + object_name
+
+            return object_name
+
+        else:
+
+            rex_obj_name = r'^.+\/(lib.+\.a)\((.+\.o)\)$'
```

Comment:
This comment is the same as the other Regular expression related ones:
1) `re` not `rex` is common in python
2) Please make this a Class or global constant